pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/DefaultAggregationExecutor.java

```diff
@@ -129,12 +133,18 @@ private void aggregateColumn(AggregationFunctionContext aggrFuncContext, Aggrega
       case AggregationFunctionFactory.DISTINCTCOUNT_AGGREGATION_FUNCTION:
       case AggregationFunctionFactory.DISTINCTCOUNTHLL_AGGREGATION_FUNCTION:
         aggregationFunction.aggregate(length, resultHolder,
-            _singleValueBlockCache.getHashCodeArrayForColumn(aggrColumn));
+            (Object) _singleValueBlockCache.getHashCodeArrayForColumn(aggrColumn));
+        break;
+
+      case AggregationFunctionFactory.FASTHLL_AGGREGATION_FUNCTION:
+        String derivedColumn = _segmentMetadata.getStarTreeMetadata().getDerivedHllColumnFromOrigin(aggrColumn);
```

Comment:
One of the review comments from Friday was to do this without startTree Metadata if the hLL columns are provided to us. You can change that as another commit but evaluate if fastHLL without starTree provides any benefit over distinctCountHLL